aarch64/src/deviceutil.rs

```diff
@@ -0,0 +1,28 @@
+use port::Result;
+use port::mem::{PhysRange, VirtRange};
+
+use crate::vm;
+
+/// Map a device register to device memory
+/// TODO Maybe make this a macro and wrap the error reporting?
+pub fn map_device_register(
```

Comment:
nit: I would call this a *register block*, since it is not just a single register, AIUI

Reply:
Fair point, I'll make the change in the next MR (so I can get this in and reduce the turnaround)

Reply:
although it's mapping more than the register block - it's mapping the page(s) enclosing the block, so I'm not sure.